src/App.tsx

```diff
@@ -1,41 +1,65 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getUsers } from './api/api';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App: React.FC = () => {
+  const [userId, setUserId] = useState(0);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [post, setPost] = useState<Post | null>(null);
```

Comment:
post is redundant in state, you should store only selected post id and load post info in components, where you display some post info